mibidata/mibi_image.py

```diff
@@ -229,10 +236,20 @@ def aperture(self):
     @aperture.setter
     def aperture(self, value):
         if value is not None and value not in APERTURE_MAP.values():
-            raise ValueError(
-                'Invalid aperture code \'{}\', must use values'
-                'from the following map: {}'.format(value, APERTURE_MAP)
-            )
+            if value in _DEPRECATED_APERTURE_MAP:
+                warnings.warn(
+                    'Deprecated aperture code \'{}\', converting to \'{}\'. In '
+                    'a future version, values from the following map will be '
+                    'required: {}'.format(value,
+                                          _DEPRECATED_APERTURE_MAP[value],
+                                          APERTURE_MAP)
+                )
+                value = _DEPRECATED_APERTURE_MAP[value]
```

Comment:
@rfinck - I just realized that you're performing the mapping in the new setter. This is actually different behavior than what was happening before. All I did was check if the passed parameter for aperture was a valid code (e.g. 'B' or 'C')